From 75ea65f79018bbb7f96e1b908cfe557c50e70b93 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Mon, 14 Dec 2020 18:07:13 +0100
Subject: [PATCH 1/1] Use Solus defaults and fixes

- Better config dir
- Use tty7 for systemd service per config
- use our /usr/share/default for xprofile and zsh
---
 CMakeLists.txt           | 4 ++--
 data/scripts/Xsession    | 3 ++-
 services/sddm.service.in | 4 ++--
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e52e0e9..1f59ccd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -183,7 +183,7 @@ endif()
 
 # Set constants
 set(DATA_INSTALL_DIR            "${CMAKE_INSTALL_FULL_DATADIR}/sddm"                CACHE PATH      "System application data install directory")
-set(DBUS_CONFIG_DIR             "${CMAKE_INSTALL_SYSCONFDIR}/dbus-1/system.d"       CACHE PATH      "DBus config files directory")
+set(DBUS_CONFIG_DIR             "${CMAKE_INSTALL_FULL_DATADIR}/dbus-1/system.d"     CACHE PATH      "DBus config files directory")
 set(STATE_DIR                   "${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/lib/sddm"      CACHE PATH      "State directory")
 set(RUNTIME_DIR                 "${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/run/sddm"      CACHE PATH      "Runtime data storage directory")
 set(QML_INSTALL_DIR             "${QT_IMPORTS_DIR}"                                 CACHE PATH      "QML component installation directory")
@@ -193,7 +193,7 @@ set(WAYLAND_SESSION_COMMAND     "${DATA_INSTALL_DIR}/scripts/wayland-session"
 
 set(CONFIG_FILE                 "${CMAKE_INSTALL_FULL_SYSCONFDIR}/sddm.conf"        CACHE PATH      "Path of the sddm config file")
 set(CONFIG_DIR                  "${CMAKE_INSTALL_FULL_SYSCONFDIR}/sddm.conf.d"      CACHE PATH      "Path of the sddm config directory")
-set(SYSTEM_CONFIG_DIR           "${CMAKE_INSTALL_PREFIX}/lib/sddm/sddm.conf.d"      CACHE PATH      "Path of the system sddm config directory")
+set(SYSTEM_CONFIG_DIR           "${CMAKE_INSTALL_PREFIX}/share/sddm/sddm.conf.d"    CACHE PATH      "Path of the system sddm config directory")
 set(LOG_FILE                    "${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/sddm.log"  CACHE PATH      "Path of the sddm log file")
 set(DBUS_CONFIG_FILENAME        "org.freedesktop.DisplayManager.conf"               CACHE STRING    "Name of the sddm config file")
 set(COMPONENTS_TRANSLATION_DIR  "${DATA_INSTALL_DIR}/translations"                  CACHE PATH      "Components translations directory")
diff --git a/data/scripts/Xsession b/data/scripts/Xsession
index 5449810..119a53f 100755
--- a/data/scripts/Xsession
+++ b/data/scripts/Xsession
@@ -21,7 +21,7 @@ case $SHELL in
     ;;
 */zsh)
     [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
-    [ -d /etc/zsh ] && zdir=/etc/zsh || zdir=/etc
+    [ -d /usr/share/defaults/etc/zsh ] && zdir=/usr/share/defaults/etc/zsh
     zhome=${ZDOTDIR:-$HOME}
     # zshenv is always sourced automatically.
     [ -f $zdir/zprofile ] && . $zdir/zprofile
@@ -50,6 +50,7 @@ case $SHELL in
     ;;
 esac
 
+[ -f /usr/share/defaults/etc/xprofile ] && . /usr/share/defaults/etc/xprofile
 [ -f /etc/xprofile ] && . /etc/xprofile
 [ -f /usr/local/etc/xprofile ] && . /usr/local/etc/xprofile
 [ -f $HOME/.xprofile ] && . $HOME/.xprofile
diff --git a/services/sddm.service.in b/services/sddm.service.in
index 0335685..932ac78 100644
--- a/services/sddm.service.in
+++ b/services/sddm.service.in
@@ -1,8 +1,8 @@
 [Unit]
 Description=Simple Desktop Display Manager
 Documentation=man:sddm(1) man:sddm.conf(5)
-Conflicts=getty@tty1.service
-After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service systemd-logind.service
+Conflicts=getty@tty7.service
+After=systemd-user-sessions.service getty@tty7.service plymouth-quit.service systemd-logind.service
 StartLimitIntervalSec=30
 StartLimitBurst=2
 
-- 
2.29.2

